[test]: CI fails on current Go tip (upcoming 1.21)

### What happened?

Unit tests fail on Go tip

### Steps to reproduce

`make test-ci` but on Go tip (e.g. `go1.21 test ./...`)

### Expected behavior

Tests should not fail.

### Relevant log output

https://github.com/jaegertracing/jaeger/actions/runs/4486863013/jobs/7889732133#step:7:1564

The difference seems to be in different types of Context argument
```diff
- context.todoCtx{emptyCtx:context.emptyCtx{}}
+ context.backgroundCtx{emptyCtx:context.emptyCtx{}}
```

<details>
<summary>Logs from run</summary>

```
--- FAIL: TestSpanProcessor_Process (0.00s)
panic: 

mock: Unexpected Method Call
-----------------------------

WriteSpan(context.todoCtx,*model.Span)
		0: context.todoCtx{emptyCtx:context.emptyCtx{}}
		1: &model.Span{TraceID:model.TraceID{Low:0x0, High:0x0}, SpanID:0x0, OperationName:"", References:[]model.SpanRef(nil), Flags:0x0, StartTime:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Duration:0, Tags:[]model.KeyValue(nil), Logs:[]model.Log(nil), Process:(*model.Process)(0xc00019c190), ProcessID:"", Warnings:[]string(nil), XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0}

The closest call I have is: 

WriteSpan(context.backgroundCtx,*model.Span)
		0: context.backgroundCtx{emptyCtx:context.emptyCtx{}}
		1: &model.Span{TraceID:model.TraceID{Low:0x0, High:0x0}, SpanID:0x0, OperationName:"", References:[]model.SpanRef(nil), Flags:0x0, StartTime:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Duration:0, Tags:[]model.KeyValue(nil), Logs:[]model.Log(nil), Process:(*model.Process)(0xc00019c190), ProcessID:"", Warnings:[]string(nil), XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0}


Diff: 0: FAIL:  (context.todoCtx=context.TODO) != (context.backgroundCtx=context.Background)
	1: PASS:  (*model.Span=trace_id:<) == (*model.Span=trace_id:<) [recovered]
	panic: 

mock: Unexpected Method Call
-----------------------------

WriteSpan(context.todoCtx,*model.Span)
		0: context.todoCtx{emptyCtx:context.emptyCtx{}}
		1: &model.Span{TraceID:model.TraceID{Low:0x0, High:0x0}, SpanID:0x0, OperationName:"", References:[]model.SpanRef(nil), Flags:0x0, StartTime:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Duration:0, Tags:[]model.KeyValue(nil), Logs:[]model.Log(nil), Process:(*model.Process)(0xc00019c190), ProcessID:"", Warnings:[]string(nil), XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0}

The closest call I have is: 

WriteSpan(context.backgroundCtx,*model.Span)
		0: context.backgroundCtx{emptyCtx:context.emptyCtx{}}
		1: &model.Span{TraceID:model.TraceID{Low:0x0, High:0x0}, SpanID:0x0, OperationName:"", References:[]model.SpanRef(nil), Flags:0x0, StartTime:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Duration:0, Tags:[]model.KeyValue(nil), Logs:[]model.Log(nil), Process:(*model.Process)(0xc00019c190), ProcessID:"", Warnings:[]string(nil), XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0}


Diff: 0: FAIL:  (context.todoCtx=context.TODO) != (context.backgroundCtx=context.Background)
	1: PASS:  (*model.Span=trace_id:<) == (*model.Span=trace_id:<)
```
</details>

--- a/cmd/ingester/app/processor/span_processor_test.go
+++ b/cmd/ingester/app/processor/span_processor_test.go
@@ -49,7 +49,7 @@ func TestSpanProcessor_Process(t *testing.T) {
 
 	message.On("Value").Return(data)
 	mockUnmarshaller.On("Unmarshal", data).Return(span, nil)
-	mockWriter.On("WriteSpan", context.Background(), span).
+	mockWriter.On("WriteSpan", context.TODO(), span).
 		Return(nil).
 		Run(func(args mock.Arguments) {
 			span := args[1].(*model.Span)

--- a/plugin/storage/es/spanstore/reader_test.go
+++ b/plugin/storage/es/spanstore/reader_test.go
@@ -356,8 +356,8 @@ func TestSpanReader_multiRead_followUp_query(t *testing.T) {
 		secondMultiSearch.On("Index", mock.AnythingOfType("string")).Return(secondMultiSearch)
 		r.client.On("MultiSearch").Return(multiSearchService)
 
-		fistMultiSearchMock := firstMultiSearch.On("Do", mock.AnythingOfType("*context.emptyCtx"))
-		secondMultiSearchMock := secondMultiSearch.On("Do", mock.AnythingOfType("*context.emptyCtx"))
+		fistMultiSearchMock := firstMultiSearch.On("Do", mock.Anything)
+		secondMultiSearchMock := secondMultiSearch.On("Do", mock.Anything)
 
 		// set TotalHits to two to trigger the follow up query
 		// the client will return only one span therefore the implementation
@@ -951,7 +951,7 @@ func mockMultiSearchService(r *spanReaderTest) *mock.Call {
 	multiSearchService.On("Index", mock.AnythingOfType("string"), mock.AnythingOfType("string"),
 		mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return(multiSearchService)
 	r.client.On("MultiSearch").Return(multiSearchService)
-	return multiSearchService.On("Do", mock.AnythingOfType("*context.valueCtx"))
+	return multiSearchService.On("Do", mock.Anything)
 }
 
 func mockArchiveMultiSearchService(r *spanReaderTest, indexName string) *mock.Call {
@@ -959,7 +959,7 @@ func mockArchiveMultiSearchService(r *spanReaderTest, indexName string) *mock.Ca
 	multiSearchService.On("Add", mock.Anything, mock.Anything, mock.Anything).Return(multiSearchService)
 	multiSearchService.On("Index", indexName).Return(multiSearchService)
 	r.client.On("MultiSearch").Return(multiSearchService)
-	return multiSearchService.On("Do", mock.AnythingOfType("*context.valueCtx"))
+	return multiSearchService.On("Do", mock.Anything)
 }
 
 // matchTermsAggregation uses reflection to match the size attribute of the TermsAggregation; neither
@@ -981,10 +981,7 @@ func mockSearchService(r *spanReaderTest) *mock.Call {
 	searchService.On("Aggregation", stringMatcher(operationsAggregation), mock.MatchedBy(matchTermsAggregation)).Return(searchService)
 	searchService.On("Aggregation", stringMatcher(traceIDAggregation), mock.AnythingOfType("*elastic.TermsAggregation")).Return(searchService)
 	r.client.On("Search", mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return(searchService)
-	return searchService.On("Do", mock.MatchedBy(func(ctx context.Context) bool {
-		t := reflect.TypeOf(ctx).String()
-		return t == "*context.valueCtx" || t == "*context.emptyCtx"
-	}))
+	return searchService.On("Do", mock.Anything)
 }
 
 func TestTraceQueryParameterValidation(t *testing.T) {

